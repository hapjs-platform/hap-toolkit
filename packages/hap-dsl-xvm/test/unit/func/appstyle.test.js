/*
 * Copyright (c) 2021-present, the hapjs-platform Project Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

const { wipeDynamic } = require('hap-dev-utils')
const { compilePage } = require('../../utils')

describe('测试全局样式', () => {
  it('全局样式正确提取', async () => {
    const stats = await compilePage('Helloworld/app.ux', 'app')
    const json = stats.toJson({source: true})
    expect(json.modules.length).toMatchInlineSnapshot(`3`)

    json.modules.forEach(module => {
      const id = wipeDynamic(module.id)
      expect(wipeDynamic(module.source)).toMatchSnapshot(id)
    })
  }, 20000)
})
