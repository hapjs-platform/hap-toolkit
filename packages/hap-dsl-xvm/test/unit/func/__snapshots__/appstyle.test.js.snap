// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试全局样式 全局样式正确提取: ../../../lib/loaders/script-loader.js!../../../../hap-packager/lib/loaders/module-loader.js!../../../../hap-packager/lib/loaders/manifest-loader.js?path=<CWD>/packages/hap-dsl-xvm/test/case/ux/Helloworld!../../../../../node_modules/babel-loader/lib/index.js?cwd=<CWD>&cacheDirectory&comments=false&configFile=<CWD>/babel.config.js!../../../lib/loaders/fragment-loader.js?index=0&type=script!./Helloworld/app.ux?uxType=app 1`] = `
"module.exports = function __scriptModule__ (module, exports, $app_require$){\\"use strict\\";

module.exports = {
  directives: {
    focus: {
      mounted(el) {
        el.focus();
      }
    }
  }
};}"
`;

exports[`测试全局样式 全局样式正确提取: ../../../lib/loaders/style-loader.js?index=0&type=style!../../../lib/loaders/fragment-loader.js?index=0&type=style!./Helloworld/app.ux?uxType=app 1`] = `
"module.exports = {
  \\".btn\\": {
    \\"color\\": \\"#ff0066\\",
    \\"backgroundColor\\": \\"#00ff00\\"
  }
}"
`;

exports[`测试全局样式 全局样式正确提取: ./Helloworld/app.ux?uxType=app 1`] = `
"
var $app_script$ = require(\\"!../../../../lib/loaders/script-loader.js!../../../../../../node_modules/@hap-toolkit/packager/lib/loaders/module-loader.js!../../../../../../node_modules/@hap-toolkit/packager/lib/loaders/manifest-loader.js?path=<CWD>/packages/hap-dsl-xvm/test/case/ux/Helloworld!../../../../../../node_modules/babel-loader/lib/index.js?cwd=<CWD>&cacheDirectory&comments=false&configFile=<CWD>/babel.config.js!../../../../lib/loaders/fragment-loader.js?index=0&type=script!./app.ux?uxType=app\\")
$app_define$('@app-component/app', [], function($app_require$, $app_exports$, $app_module$) {
     $app_script$($app_module$, $app_exports$, $app_require$)
        if ($app_exports$.__esModule && $app_exports$.default) {
          $app_module$.exports = $app_exports$.default
        }
    $app_module$.exports.template = {}
    $app_module$.exports.style = require(\\"!../../../../lib/loaders/style-loader.js?index=0&type=style!../../../../lib/loaders/fragment-loader.js?index=0&type=style!./app.ux?uxType=app\\")
});
;"
`;
